scripts/import_users.py

```diff
@@ -0,0 +1,153 @@
+import pandas as pd
+
+# modin enables using multiple cores for pandas but is "not yet optimized" for all groupby operations,
+# and isn't optimized for the groupby operation used in this script
+
+# Options to set before running the script
+use_subset_of_data = False  # Set to True if you're developing and want operations to take less time
+generate_from_source = True  # Set to True to regenerate intermediate data first, vs loading saved intermediate data
+
+# Increase amount of data displayed in terminal
+pd.set_option('display.max_rows', 500)
+pd.set_option('display.max_columns', 200)
+pd.set_option('display.width', 1000)
+
+HASH_LENGTH = 66
+VALID_WEEKS = [
+    "2020-02-08",
+    "2020-02-15",
+    "2020-02-22",
+    "2020-02-29",
+    "2020-03-07",
+    "2020-03-14",
+    "2020-03-21",
+    "2020-03-28",
+    "2020-04-04",
+    "2020-04-11",
+    "2020-04-18",
+    "2020-04-25",
+    "2020-05-02",
+    "2020-05-09"
+]
+WEEKS_TO_INDEX = dict(map(reversed, enumerate(VALID_WEEKS)))  # { "2020-04-15": "0" ... }
+INDEX_TO_WEEKS = dict(enumerate(VALID_WEEKS))  # { "0": "2020-04-15"... }
+VALID_PROGRAMS = ["DUA", "UI"]
+VALID_PLANS = ["UI part time", "UI full time", "PUA full time"]
+SOURCE_DATA_FILENAME = "users.csv"
+INTERMEDIATE_DATA_FILENAME = "intermediate.pkl"
+INTERMEDIATE_DATA_100K_FILENAME = "100k.pkl"  # Generate a smaller file of 100k rows during development process
+DUPLICATE_HASHES_FILENAME = "duplicate_hashes.json"
+FINAL_DATA_100K_FILENAME = "100k.json"
+FINAL_DATA_FILENAME = "users.json"
+
+intermediate_filename = INTERMEDIATE_DATA_FILENAME
+final_filename = FINAL_DATA_FILENAME
+if use_subset_of_data:
+    intermediate_filename = INTERMEDIATE_DATA_100K_FILENAME
+    final_filename = FINAL_DATA_100K_FILENAME
+
+
+def generate_final_file():
+    def print_rows_with_week_count(df, count):
+        print(df.loc[df["week_count"] == count])
+
+    # Print the number of users who need to certify for a total of X weeks, 1 through 14
+    # # of weeks  # of users who must certify
+    #           1                  19280
+    #           2                  808167
+    def print_total_weeks_by_user(df):
+        df["week_count"] = df["WeekEndingDates"].apply(lambda x: len(x))  # Count number of weeks per user
+        series = df["week_count"].value_counts(normalize=True).sort_index().mul(100).round(1).astype(str) + '%'
+        result = pd.DataFrame(series).reset_index()
+        result.columns = ["# of weeks", "% of users who must certify"]
+        print(result.to_string(index=False))
+
+    def print_duplicate_hashes(df):
+        # df.pivot_table(index=["SHA256_hash"], aggfunc="size").sort_values()
+        dupe_hashes = df[df.duplicated("SHA256_hash", keep=False)]
+        dupe_hashes_count = len(dupe_hashes)
+        print(f"There are {dupe_hashes_count} duplicate hashes:")
+        print(dupe_hashes)
+
+    print(f"Importing {intermediate_filename}. Delete it to import {intermediate_filename} instead")
```

Comment:
I don't understand this comment. Delete it to do what?

Reply:
Ah it's an outdated comment back when I was checking file existence, I'll remove